### ISSUE TITLE
Google Third Party Login doesn't log you in

### ISSUE THREAD

**Issue by Phinalize** [streetcomplete/StreetComplete#5711]
**Description**
I'm a new user looking to contribute, I've been asked to log into the app. When I choose the Google alternative, it opens the Google authentication page where it asks for the Email address and credential, however, once these have been provided the app reverts back to the Map but I'm not logged in.

**How to Reproduce**
1. Open the StreetComplete app
2. Click on the Hamburger menu and select "My Profile"
3. Click on "Login"
4. "Log in to OpenStreetMap to access StreetComplete" window should appear, choose the Google image under "Alternatively, use a third party to login" section towards the bottom.
5. Provide necessary details for the Google account, confirming any MFA actions (for me, this is a 2-step using the Device I have StreetComplete installed on)
6. You're taken back to the map, but are not logged in (checking via Hamburger Menu > My Profile)

**Expected Behavior**
Profile to be created and to be signed in.

**Versions affected**
Android 14 (Build AP2A.240605.024), StreetComplete v58.0

**Comment by matkoniecz** [comment]
this steps should result in your OpenStreetMap account being created (associated with Google account).

What happens when you try to log in then?


(yes, even if it works it is not working in a proper way but it would be interesting how exactly it is broken)

**Comment by Phinalize** [comment]
I've sent a crash report from super**********n@gmail.com, if that helps.

**Comment by Phinalize** [comment]
> this steps should result in your OpenStreetMap account being created (associated with Google account).
> 
> What happens when you try to log in then?
> 
> (yes, even if it works it is not working in a proper way but it would be interesting how exactly it is broken)

I can log into OpenStreetMap using my Google account without issue, only seems to be a problem when done via StreetComplete

**Comment by westnordost** [comment]
Thank you, here is the crash report (it doesn't contain private information):

```
java.lang.IllegalStateException: Fragment LoginFragment{13bf768} (b0d36a4b-1464-42ea-a950-79a79a96f06c) did not return a View from onCreateView() or this was called before onCreateView().
at androidx.fragment.app.Fragment.requireView(Unknown Source:31)
at de.westnordost.streetcomplete.util.FragmentViewBindingPropertyDelegate.getValue(SourceFile:0)
at de.westnordost.streetcomplete.screens.user.login.LoginFragment.getBinding(Unknown Source:7)
at de.westnordost.streetcomplete.screens.user.login.LoginFragment.access$getBinding(Unknown Source:0)
at de.westnordost.streetcomplete.screens.user.login.LoginFragment$OAuthWebViewClient.onPageStarted(Unknown Source:7)
at WV.xZ.c(chromium-TrichromeWebViewGoogle6432.apk-stable-622500032:16)
at WV.D6.handleMessage(chromium-TrichromeWebViewGoogle6432.apk-stable-622500032:505)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loopOnce(Looper.java:232)
at android.os.Looper.loop(Looper.java:317)
at android.app.ActivityThread.main(ActivityThread.java:8592)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
```

I remember @Helium314 mentioning this to me but I not being able to reproduce this, I then discarded that as "probably some Android weirdness/bug".

Looking at this crash report now again in the google play console (where crash reports are aggregated automatically), I see a lot of people do have this issue, uff. I should have checked that better. And, the time this issue first occurred seems to align with a refactor I did on the LoginFragment, namely introducing a ViewModel.

I will investigate this further ASAP.

**Comment by Phinalize** [comment]
Excellent, thank you for looking into it

**Comment by Phinalize** [comment]
> Thank you for the hint that you use Google for login, will try to get it reproduced that way in case I continue to not be able to reproduce this with a normal login. When using a Google Account, are you lead somehow outside of the app and into your browser?

I remain in the app up until the 2FA confirmation, which takes me out of the app but will take me back once confirmed

**Comment by westnordost** [comment]
Oh, actually, I think the problem with Google and the crash were two separate problems. The login screen would just re-start everytime one reenters the app. Anyway, will be fixed with #5727
